src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars "
+                + "SET is_deleted = TRUE "
+                + "WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete the car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id, manufacturer_id, model "
+                + "FROM cars "
+                + "JOIN cars_drivers cd "
+                + "ON cars.id = cd.car_id "
+                + "WHERE cd.driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> carsList = new ArrayList<>();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                carsList.add(car);
+            }
+            return carsList;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't get all driver's cars by id " + driverId, e);
+        }
+    }
+
+    private void addDriversToCar(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            List<Driver> drivers = car.getDrivers();
+            for (Driver driver : drivers) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't insert new cars_drivers relationship by " + car, e);
+        }
+    }
+
+    private void deleteDriversFromCar(Long carId) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete drivers for car with id " + carId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        String query = "SELECT id FROM manufacturers WHERE id = ?";
```

Comment:
you don't need a separate query for getting this information, you should get this information from query for cars